BigQuery/13_iv_antibiotics_pharmacy.sql

```diff
@@ -0,0 +1,86 @@
+min(entertime) as entertime_min,
+min(starttime) as starttime_min,
+min(route) as route_min,
+max(stoptime) as stoptime_max,
+min(frequency) as frequency_min,
+min(disp_sched) as disp_sched_min,
+min(doses_per_24_hrs) as doses_per_24_hrs_min,
+min(duration) as duration_min,
+min(duration_interval) as duration_interval_min,
+count(1) as records
+from `physionet-data.mimic_hosp.pharmacy`
+JOIN `physionet-data.mimic_icu.icustays`
+USING (subject_id, hadm_id)
+where
+(
+    --medical abx names
+    lower(medication) like "%vancomycin%"
+    or lower(medication) like "%ceftriaxone%"
+    or lower(medication) like "%meropenem%"
+    or lower(medication) like "%ceftazidime%"
+    or lower(medication) like "%cefotaxime%"
+    or lower(medication) like "%cefepime%"
+    or lower(medication) like "%piperacillin%"
+    or lower(medication) like "%tazobactam%"
+    or lower(medication) like "%amipcillin%"
+    or lower(medication) like "%sulbactam%"
+    or lower(medication) like "%imipenem%"
+    or lower(medication) like "%cilastatin%"
+    or lower(medication) like "%levofloxacin%"
+    or lower(medication) like "%clindamycin%"
+    or lower(medication) like "%metronidazol%"
+    or lower(medication) like "%linezolid%"
+    or lower(medication) like "%ciprofloxacin%"
+    or lower(medication) like "%nafcillin%"
+    or lower(medication) like "%daptomycin%"
+    or lower(medication) like "%amikacin%"
+    or lower(medication) like "%tobramycin%"
+    or lower(medication) like "%gentamicin"
+    --brand names
+    or lower(medication) like "%vancocin%"
+    or lower(medication) like "%rocephin%"
+    or lower(medication) like "%merrem%"
+    or lower(medication) like "%fortaz%"
+    or lower(medication) like "%claforan%"
+    or lower(medication) like "%maxipime%"
+    or lower(medication) like "%zosyn%"
+    or lower(medication) like "%unasyn%"
+    or lower(medication) like "%primaxin%"
+    or lower(medication) like "%levaquin%"
+    or lower(medication) like "%cleocin%"
+    or lower(medication) like "%flagyl%"
+    or lower(medication) like "%zyvox%"
+    or lower(medication) like "%cetraxal%"
+    or lower(medication) like "%cipro%"
+    or lower(medication) like "%otiprio%"
+    or lower(medication) like "%ciloxan%"
+    or lower(medication) like "%cubicin%"
+    or lower(medication) like "%tobi podhaler%"
+    or lower(medication) like "%bethkis%"
+    or lower(medication) like "%tobrex%"
+    or lower(medication) like "%gentak%"
+)
+and route in (
+    'IV' --this captures by far the largest portion of abx orders, followed with a few from IVIC and IVT. The others are here for posterity
+    ,'IVCI'
+    ,'IVIC'
+    ,'IVS'
+    ,'IVT')
+AND
+lower(medication) not like "%desensitization%" --ignore desensitization protocols in preparation for abx
+AND
+DATETIME_TRUNC(starttime, HOUR) < outtime 
+AND
+DATETIME_TRUNC(starttime, HOUR) > intime
+GROUP BY subject_id, hadm_id, stay_id, intime, outtime, chart_hour, medication
```

Comment:
The extra `group by` values here are confusing again. 

- Of subject_id, hadm_id, and stay_id, my concept of reality is that only stay_id is needed.
- Neither intime nor outtime should be needed on the `group by` because they're already associated with the stay.
- Grouping by chart hour may be ok, but some values, like `doses_per_24_hrs_min`, may be broken. I'm not sure.
- The `medication` column makes sense to include.

Reply:
Good point, the intime and outtime fields should probably just be removed from the query altogether, since that's coming from the icustays table - I think I kept those in there for validation purposes (to compare the intime and outtime of the stay vs the abx order timing) and never got rid of it

Reply:
Ok, sounds good!